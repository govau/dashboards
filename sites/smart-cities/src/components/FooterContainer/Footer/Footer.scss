@import '~style/main';

.wrapper {
  padding: 50px 0 55px;
  background: $color-grey--800;
  color: $color-white;
}

.container {
  @include container;

  display: flex;
  flex-direction: column;
  align-items: center;

  @include for-desktop-up {
    flex-direction: row;
    align-items: flex-start;
  }
}

.logosWrapper {
  display: flex;
  align-items: center;
  flex: none;
  margin-bottom: 50px;

  @include for-desktop-up {
    margin-bottom: 0;
    margin-right: 80px;
  }
}

.coaWrapper {
  margin-right: 18px;
}

.coaLogo {
  width: 120px;
  height: auto;
}

.logoDivider {
  width: 2px;
  height: 46px;
  margin-right: 32px;
  background: $color-grey--700;
}

.dtaLogo {
  width: 70px;
  height: auto;
}

.wordsWrapper {
  flex: auto;
}

.categoryDetailsWrapper {
  border-bottom: 1px solid $color-grey--700;
  margin-bottom: 20px;
}

.categoryTitle {
  margin-bottom: 16px;
  font-size: 20px;
  font-weight: 700;
}

.subCategoryLinksWrapper {
  display: flex;
  align-items: flex-start;
  flex-direction: column;
  margin-bottom: 23px;

  @include for-tablet-portrait-up {
    flex-direction: row;
  }
}

.subCategoryLink {
  margin-right: 32px;
  text-decoration: none;
  padding-bottom: 1px;
  border-bottom: 1px solid $color-highlight;
  margin-bottom: 16px;

  @include for-tablet-portrait-up {
    margin-bottom: 0;
  }
}

.staticLinksWrapper {
  margin-bottom: 6px;
}

.staticLink {
  margin-right: 16px;
  font-size: 14px;
  text-decoration: underline;
}

.contact {
  border-bottom: 1px solid $color-grey--700;
  font-size: 14px;
  padding-bottom: 24px;
  margin-bottom: 24px;
}

.contactLink {
  text-decoration: underline;
}

.copyright {
  font-size: 14px;
  line-height: 1.2;
}
