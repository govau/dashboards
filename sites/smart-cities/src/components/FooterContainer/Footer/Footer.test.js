import React from 'react';
import { shallow } from 'enzyme';
import Footer from './Footer';

it('should match Snapshot', () => {
  const component =  shallow(<Footer />);
  expect(component.debug()).toMatchSnapshot();
});

it('should not render the category section if none is passed', () => {
  const component =  shallow(<Footer />);

  expect(component.find('.categoryDetailsWrapper').length).toBe(0);
});

it('should render the category section if passed categories', () => {
  const component =  shallow(
    <Footer
      category={{
        name: 'The category',
        subCategories: [
          { name: 'sub-category one' },
          { name: 'sub-category two' },
          { name: 'sub-category three' },
        ]
      }}
    />
  );

  expect(component.find('.categoryDetailsWrapper').length).toBe(1);
  expect(component.find('.subCategoryLink').length).toBe(3);
});
