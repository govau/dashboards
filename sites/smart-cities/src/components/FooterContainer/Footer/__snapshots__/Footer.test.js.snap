// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match Snapshot 1`] = `
"<footer className=\\"wrapper\\">
  <div className=\\"container\\">
    <div className=\\"logosWrapper\\">
      <div className=\\"coaWrapper\\">
        <img className=\\"coaLogo\\" src=\\"coa-white.svg\\" alt=\\"coat of arms\\" />
      </div>
      <div className=\\"logoDivider\\" />
      <img className=\\"dtaLogo\\" src=\\"dtaLogo.svg\\" alt=\\"dta\\" />
    </div>
    <div className=\\"wordsWrapper\\">
      <div className=\\"footerLinks\\">
        <div className=\\"footerLinksRow\\">
          <NavLink to=\\"/feedback\\" className=\\"staticLink\\" activeClassName=\\"active\\" ariaCurrent=\\"true\\">
            Give feedback
          </NavLink>
          <NavLink to=\\"/additional-information#privacy\\" className=\\"staticLink\\" activeClassName=\\"active\\" ariaCurrent=\\"true\\">
            Privacy policy
          </NavLink>
          <NavLink to=\\"/additional-information#accessibility\\" className=\\"staticLink\\" activeClassName=\\"active\\" ariaCurrent=\\"true\\">
            Accessibility
          </NavLink>
          <NavLink to=\\"/additional-information#security\\" className=\\"staticLink\\" activeClassName=\\"active\\" ariaCurrent=\\"true\\">
            Security
          </NavLink>
          <NavLink to=\\"/additional-information#content-credit\\" className=\\"staticLink\\" activeClassName=\\"active\\" ariaCurrent=\\"true\\">
            Content credit
          </NavLink>
        </div>
        <div className=\\"footerLinksRow\\">
          Got a question on the Performance Framework? Contact PM&amp;C
          : 
          <a href=\\"mailto:cityperformance@pmc.gov.au?subject=Performance Framework\\">
            cityperformance@pmc.gov.au
          </a>
        </div>
        <div className=\\"footerLinksRow\\">
          <Link href=\\"https://data.gov.au/dataset/national-cities-performance-framework\\" unstyled={true}>
            View all raw data at data.gov.au
          </Link>
        </div>
      </div>
      <div className=\\"copyright\\">
        Â© Commonwealth of Australia. With the exception of the Commonwealth Coat of Arms and where otherwise noted, this work is licensed under the CC BY 4.0 license.
      </div>
    </div>
  </div>
</footer>"
`;
